spec/immutable_struct_spec.rb

```diff
@@ -103,14 +103,28 @@ def self.from_array(array)
   end
 
   describe "to_h" do
-    it "should include the output of params and block methods in the hash" do
+    it "should not include the output of params and block methods in the hash" do
       klass = ImmutableStruct.new(:flappy) do
         def lawsuit
           'pending'
         end
       end
       instance = klass.new(flappy: 'bird')
-      instance.to_h.should == {flappy: 'bird', lawsuit: 'pending'}
+      instance.to_h.should == {flappy: 'bird'}
     end
+
+    it "allows #== to be overriden" do
```

Comment:
The issue isn't with the method `==` is it?  It's that nonzero-arity methods are included in the `to_h` implementation.

So, I think this test is mostly fine, but I'd change this line to somethingn like `it "does not include nonzero arity methods in the output"`